src/app/modules/ix-table2/interfaces/table-column.interface.ts

```diff
@@ -26,6 +28,6 @@ export abstract class ColumnComponent<T> {
 }
 
 export type Column<T, C extends ColumnComponent<T>> = {
-  type?: new (matDialog?: MatDialog) => C;
+  type?: new (matDialog?: MatDialog, translate?: TranslateService, dialogService?: DialogService) => C;
```

Comment:
Following this approach does not looks good. For every specific need it is extending common component. Maybe [inject](https://angular.io/api/core/inject) function can help to get necessary services in the state button component and keep this clean?